sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,175 @@
+import { credentialLogger, formatSuccess, formatError } from "../util/logging";
+import { trace } from "../util/tracing";
+import { ensureValidScope, getScopeResource } from "../util/scopeUtils";
+
+const logger = credentialLogger("AzurePowerShellCredential");
+
+/**
+ * Formats a command based on the platform executing it.
+ * @internal
+ */
+function formatCommand(commandName: string): string {
+  if (process.platform === "win32") {
+    return `${commandName}.exe`;
+  } else {
+    return commandName;
+  }
+}
+
+/**
+ * Receives a list of commands to run, executes them, then returns the outputs.
+ * If anything fails, an error is thrown.
+ * @internal
+ */
+function runCommands(commands: string[]): string[] {
+  const result: string[] = [];
+
+  try {
+    commands.forEach((cmd: string) => {
+      result.push(childProcess.execSync(cmd).toString("utf-8"));
+    });
+  } catch (e) {
+    throw new Error(e.toString("utf-8"));
+  }
+
+  return result;
+}
+
+/**
+ * Known PowerShell errors
+ * @internal
+ */
+export const powerShellErrors = {
+  login: "Run Connect-AzAccount to login",
+  installed:
+    "The specified module 'Az.Accounts' with version '2.2.0' was not loaded because no valid module file was found in any module directory"
+};
+
+/**
+ * Messages to use when throwing in this credential.
+ * @internal
+ */
+export const powerShellPublicErrorMessages = {
+  login:
+    "Please run 'Connect-AzAccount' from powershell to authenticate before using this credential.",
+  installed: `Az.Account module >= 2.2.0 is not installed. Install the Azure Az PowerShell module with: "Install - Module - Name Az - Scope CurrentUser - Repository PSGallery - Force".`
+};
+
+// PowerShell Azure User not logged in error check.
+const isLoginError = (err: Error) => err.message.match(`(.*)${powerShellErrors.login}(.*)`);
+
+// Az Module not Installed in Azure PowerShell check.
+const isNotInstallError = (err: Error) => err.message.match(powerShellErrors.installed);
+
+/**
+ * Optional parameters for the {@link AzurePowerShellCredential} class.
+ */
+export interface AzurePowerShellCredentialOptions {
+  /**
+   * If specified, this credential will use the legacy `powershell` command instead of the newest `pwsh`.
+   */
+  useLegacyPowerShell?: boolean;
+}
+
+/**
+ * This credential will use the currently logged-in user login information via the Azure Power Shell command line tool.
+ * To do so, it will read the user access token and expire time with Azure Power Shell command `Get-AzAccessToken -ResourceUrl {ResourceScope}`.
+ * To be able to use this credential, ensure that:
+ * - Install the Azure Az PowerShell module with: `Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force`.
+ * - You have already logged in via the 'az' tool using the command "Connect-AzAccount" from the command line.
+ */
+export class AzurePowerShellCredential implements TokenCredential {
+  private useLegacyPowerShell: boolean;
```

Comment:
I would just make `powershellCommand` a private readonly string property in this case. No need to store the option, just use the option in the constructor to pick the powershell command.